web/ui/ui.go

```diff
@@ -0,0 +1,204 @@
+// Package ui exposes utilities to get a Handler for the Grafana Agent Flow UI.
+package ui
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"path"
+	"strings"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/common/server"
+)
+
+// RegisterRoutes registers routes to the provided mux.Router for serving the
+// Grafana Agent Flow UI. The UI will be served relative to pathPrefix. If no
+// pathPrefix is specified, the UI will be served at root.
+//
+// By default, the UI is retreived from the ./web/ui/build directory relative
+// to working directory, assuming that the Agent is run from the repo root.
+// However, if the builtinassets Go tag is present, the built UI will be
```

Comment:
Is there a reason we offer both ways? For ease of debugging?

Reply:
Yeah, debugging is one benefit, but it also simplifies the build process for people who aren't working with the UI, since otherwise you need NodeJS and Yarn installed just to build the agent at all.